boards/PDM/Src/Io/Io_Efuse.c

```diff
@@ -0,0 +1,935 @@
+    ((0x01 << CSNS1_EN_SHIFT) | (0x00 << CSNS0_EN_SHIFT))
+#define CSNS_FUNCTION_TEMPERATURE \
+    ((0x01 << CSNS1_EN_SHIFT) | (0x01 << CSNS0_EN_SHIFT))
+#define CSNS_FUNCTION_CURRENT_SUM \
+    ((0x00 << CSNS1_EN_SHIFT) | (0x01 << CSNS0_EN_SHIFT))
+
+#define VDD_FAILURE_DETECTION_ENABLE (0x01 << VDD_FAIL_EN_SHIFT)
+#define VDD_FAILURE_DETECTION_DISABLE (0x00 << VDD_FAIL_EN_SHIFT)
+
+#define WATCHDOG_DISABLE (0x01 << WD_DIS_SHIFT)
+#define WATCHDOG_ENABLE (0x00 << WD_DIS_SHIFT)
+
+#define TRACK_AND_HOLD_CURRENT_ENABLE (0x01 << T_H_EN_SHIFT)
+#define TRACK_AND_HOLD_CURRENT_DISABLE (0x00 << T_H_EN_SHIFT)
+
+#define PARALLEL_MODE_ENABLE (0x01 << PARALLEL_SHIFT)
+#define PARALLEL_MODE_DISABLE (0x00 << PARALLEL_SHIFT)
+
+#define PWM_CHANNEL_0_ENABLE (0x01 << PWM_EN_0_SHIFT)
+#define PWM_CHANNEL_0_DISABLE (0x00 << PWM_EN_0_SHIFT)
+
+#define PWM_CHANNEL_1_ENABLE (0x01 << PWM_EN_1_SHIFT)
+#define PWM_CHANNEL_1_DISABLE (0x00 << PWM_EN_1_SHIFT)
+
+// Configuration values used for initializing the registers
+
+// Global Configuration
+#define GCR_CONFIG                                                           \
+    (PWM_CHANNEL_0_DISABLE | PWM_CHANNEL_1_DISABLE | PARALLEL_MODE_DISABLE | \
+     TRACK_AND_HOLD_CURRENT_DISABLE | WATCHDOG_DISABLE |                     \
+     VDD_FAILURE_DETECTION_ENABLE | CSNS_FUNCTION_DISABLED |                 \
+     OVERVOLTAGE_PROTECTION_ENABLE)
+
+// Channel  Configurations
+#define RETRY_CONFIG                                               \
+    (RANDOM_CURRENT_OFFSET_SUBTRACT | AUTO_RETRY_NUMBER_INFINITE | \
+     AUTO_RETRY_PERIOD_T_AUTO_10 | AUTO_RETRY_DC_MOTOR_DISABLED)
+#define CONFR_CONFIG                                                  \
+    (SHORT_CIRCUIT_DETECTION_ENABLE | OPEN_LOAD_DETECTION_ON_ENABLE | \
+     OPEN_LOAD_DETECTION_OFF_ENABLE | DIRECT_CONTROL_ENABLE |         \
+     SWITCH_ON_DELAY_NO_DELAY | SLEW_RATE_MEDIUM)
+#define OCR_LOW_CURRENT_SENSE_CONFIG                              \
+    (PWM_EXTERNAL_CLOCK_DIVIDER_256 | PWM_CLOCK_SOURCE_EXTERNAL | \
+     THRESHOLD_ACTIVATION_TIMES_T_OCH1_T_OCM1_L |                 \
+     MEDIUM_CURRENT_THRESHOLD_SELECT_IOCM2 |                      \
+     LOW_CURRENT_THRESHOLD_SELECT_IOCL2 |                         \
+     HIGH_CURRENT_THRESHOLD_SELECT_IOCH2 | CURRENT_SENSE_RATIO_LOW)
+#define OCR_HIGH_CURRENT_SENSE_CONFIG                             \
+    (PWM_EXTERNAL_CLOCK_DIVIDER_256 | PWM_CLOCK_SOURCE_EXTERNAL | \
+     THRESHOLD_ACTIVATION_TIMES_T_OCH1_T_OCM1_L |                 \
+     MEDIUM_CURRENT_THRESHOLD_SELECT_IOCM2 |                      \
+     LOW_CURRENT_THRESHOLD_SELECT_IOCL3 |                         \
+     HIGH_CURRENT_THRESHOLD_SELECT_IOCH2 | CURRENT_SENSE_RATIO_HIGH)
+
+struct EfuseIo
+{
+    // The current state of the watchdog-in bit (bit 15). If the watchdog is
+    // enabled its state must be alternated at least once within the watchdog
+    // timeout period.
+    bool wdin_bit_to_set;
+};
```

Comment:
@pasitlaothamatas  do you have a better name for this or is this OK? Can't call it struct Efuse since there is a namespace conflic with App_Efuse.h
https://github.com/UBCFormulaElectric/Consolidated-Firmware/blob/61026dc911d41ab4e2029a5a6248d29eab477585/boards/PDM/Inc/App/App_Efuse.h#L36

Reply:
This is also the struct where we would include the members for the shared_spi driver if we end up making one

Reply:
For the Io layer I think we name the struct `Efuse_Context`. As for the shared spi driver I'm not too sure. Lets discuss soon though

Reply:
RENAMED

Reply:
you sure are influential @kirons96 .... you sure are influential

Reply:
@pabsfdez KUMRON SEN is my SIMP QUEEN